internal/controllers/common_controller.go

```diff
@@ -0,0 +1,119 @@
+}
+
+// setResolvedStatusConditionSuccess sets the resolved status condition to success.
+func setResolvedStatusConditionSuccess(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeResolved,
+		Status:             metav1.ConditionTrue,
+		Reason:             ocv1alpha1.ReasonSuccess,
+		Message:            message,
+		ObservedGeneration: generation,
+	})
+}
+
+// setInstalledStatusConditionUnknown sets the installed status condition to unknown.
+func setInstalledStatusConditionUnknown(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeInstalled,
+		Status:             metav1.ConditionUnknown,
+		Reason:             ocv1alpha1.ReasonInstallationStatusUnknown,
+		Message:            message,
+		ObservedGeneration: generation,
+	})
+}
+
+// setResolvedStatusConditionFailed sets the resolved status condition to failed.
+func setResolvedStatusConditionFailed(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeResolved,
+		Status:             metav1.ConditionFalse,
+		Reason:             ocv1alpha1.ReasonResolutionFailed,
+		Message:            message,
+		ObservedGeneration: generation,
+	})
+}
+
+// setResolvedStatusConditionUnknown sets the resolved status condition to unknown.
+func setResolvedStatusConditionUnknown(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeResolved,
+		Status:             metav1.ConditionUnknown,
+		Reason:             ocv1alpha1.ReasonResolutionUnknown,
+		Message:            message,
+		ObservedGeneration: generation,
+	})
+}
+
+// setInstalledStatusConditionSuccess sets the installed status condition to success.
+func setInstalledStatusConditionSuccess(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeInstalled,
+		Status:             metav1.ConditionTrue,
+		Reason:             ocv1alpha1.ReasonSuccess,
+		Message:            message,
+		ObservedGeneration: generation,
+	})
+}
+
+// setInstalledStatusConditionFailed sets the installed status condition to failed.
+func setInstalledStatusConditionFailed(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeInstalled,
+		Status:             metav1.ConditionFalse,
+		Reason:             ocv1alpha1.ReasonInstallationFailed,
+		Message:            message,
+		ObservedGeneration: generation,
+	})
+}
+
+func setDeprecationStatusesUnknown(conditions *[]metav1.Condition, message string, generation int64) {
```

Comment:
every other function got a comment, except for me!

Reply:
That's the way it was in the original code (I'm guessing).